### PR TITLE
[#373] Update all of our old unit tests. Purge the repo of waste.

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/373-bring-unit-tests-back-up-to-par

### CODE REVIEW

File: test/comm/comm.unit.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require( [ "../src/core/comm" ], function( Comm ) {
```
Comment (secretrobotron) [inline test/comm/comm.unit.js]: This is _actually_ nothing that uses `comm` anymore. Nothing. I checked. I would be ok with removing it and its tests entirely. :)

File: test/media/media.unit.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    Butter({
+      config: "../test-config-core.json",
+      debug: false,
+      ready: function( butter ){
+        callback( butter );
+      }
+    });
+  }
+  module( "Media" );
```
Comment (secretrobotron) [inline test/media/media.unit.js]: Mind adding a newline above this line?

File: test/page/expectedPage.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<html><head><base href="http://localhost:8888/test/page/"><script async="" src="http://s.ytimg.com/yt/jsbin/www-widgetapi-vflhJ8ZTS.js"></script>
+    <title>Basic template</title>
+    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
+
```
Comment (secretrobotron) [inline test/page/expectedPage.html]: There's a bunch of whitespace here that we don't need.

File: test/tests.js [hunk 4]
```diff
@@ -5,15 +5,43 @@ var Butter_tests = [{
...
 }, {
   name: "Logger",
   path: "logger/index.html",
-  type: "Error Logging"
+  type: "Core Modules"
 }, {
   name: "Cornfield",
   path: "cornfield/index.html",
   type: "Cornfield"
+}, {
```
Comment (secretrobotron) [inline test/tests.js]: Do you mind fixing the spacing for these brackets? Nothing else in our code bases uses this style.

File: test/track/track.unit.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
+(function(window, document, undefined ){
```
Comment (secretrobotron) [inline test/track/track.unit.js]: Add some spacing here before `window`.

Comment (secretrobotron) [inline test/track/track.unit.js]: Actually, do we even need this outer scope to include `window`, `document` and `undefined`?

File: test/trackevent/trackevent.unit.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
+(function(window, document, undefined ){
```
Comment (secretrobotron) [inline test/trackevent/trackevent.unit.js]: Same as above. Probably don't really need these, do we?

File: test/page/expectedPage.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+          font-style: bold;
+        }
+        #main {
+          min-width: 300px;
+          min-height: 200px;
+        }
+    </style>
+<link href="../../css/butter.ui.css" rel="stylesheet"><script type="text/javascript" src="../../external/popcorn-js/popcorn.js"></script><script type="text/javascript" src="../../external/popcorn-js/modules/player/popcorn.player.js"></script><script src="../../external/popcorn-js/plugins/attribution/popcorn.attribution.js"></script><script src="../../external/popcorn-js/plugins/webpage/popcorn.webpage.js"></script><script src="../../external/popcorn-js/plugins/text/popcorn.text.js"></script><script src="../../external/popcorn-js/plugins/googlemap/popcorn.googlemap.js"></script><script src="../../external/popcorn-js/plugins/image/popcorn.image.js"></script><script src="../../external/popcorn-js/plugins/twitter/popcorn.twitter.js"></script><script src="../../external/popcorn-js/plugins/wikipedia/popcorn.wikipedia.js"></script><script type="text/javascript" src="../../external/popcorn-js/players/youtube/popcorn.youtube.js"></script></head><body class="butter-header-spacing butter-tray-spacing butter-editor-spacing">
```
Comment (secretrobotron) [inline test/page/expectedPage.html]: This spacing is definitely wrong :P.

File: test/page/expectedPage.html [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        <div class="content-div" id="Area2"></div>
+        <p>Area 2</p>
+    </div>
+    <div id="target-div"></div>
+
+
+<div id="editor-area" class="butter-editor-area minimized">
+<div title="Show/Hide Editor" class="butter-toggle-button toggled">
+  <div class="image-container"></div>
+</div></div></body></html>
```
Comment (secretrobotron) [inline test/page/expectedPage.html]: This spacing too.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/00bfce92becdf14/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/00bfce92becdf14/output.txt

Total script time: 1.45 mins

Server is running on http://pr629.butyr.org/templates/test.html

File: test/butter.testrunner.js [hunk 9]
```diff
@@ -1,151 +1,228 @@
...
+        results_arr = [],
+        currentTest,
+        testList = [],
+        results_arr = [],
+        userAgent = id( "qunit-userAgent" );
+
+
+    if ( userAgent ) {
+      userAgent.innerHTML = navigator.userAgent;
+    };
```
Comment (secretrobotron) [inline test/butter.testrunner.js]: This semicolon is unnecessary, right?

File: test/butter.testrunner.js [hunk 10]
```diff
@@ -1,151 +1,228 @@
...
+
+    function sendGetFocus( event ) {
+      event.target &&
+        event.target.contentWindow &&
+          event.target.contentWindow.postMessage( "getFocus", "*" );
```
Comment (secretrobotron) [inline test/butter.testrunner.js]: Do you mind making this into an `if` block? It's grown past the one-line level of acceptability, I think :P.

File: test/butter.testrunner.js [hunk 11]
```diff
@@ -1,151 +1,228 @@
...
+          return document.createElement( type );
+        },
+        index = 0,
+        testFrame = id( "test-frame" ),
+        results = id( "qunit-tests" ),
+        totalPass = 0,
+        totalFail = 0,
+        totalRun = 0,
+        totalTime = 0,
+        main_li = create( "li" ),
```
Comment (secretrobotron) [inline test/butter.testrunner.js]: These used to be properly camel-cased. Can you make them like that again for conformity's sake?

File: test/butter.testrunner.js [hunk 12]
```diff
@@ -1,151 +1,228 @@
...
+
+              anchor = document.createElement( "a" );
+              anchor.target = "_blank";
+              anchor.href = "../" + testGroup[ f ];
+              anchorText = document.createTextNode( testName );
+              anchor.appendChild( anchorText );
+              testLinks.appendChild( anchor );
+            }
+          }
+          loadedCallback && typeof loadedCallback === "function" && loadedCallback();
```
Comment (secretrobotron) [inline test/butter.testrunner.js]: Can you make an `if` block out of this line too?

File: test/butter.testrunner.js [hunk 13]
```diff
@@ -1,151 +1,228 @@
...
+      }
+    };
+
+    // Tells the tests within the iframe to take focus
+    testFrame.addEventListener( "load", sendGetFocus, false );
+
+    // Populate the userAgent h2 with information, if available
+    if ( userAgent ) {
+      userAgent.innerHTML = navigator.userAgent;
+    };
```
Comment (secretrobotron) [inline test/butter.testrunner.js]: Another unneeded semi-colon.

File: test/page/page.unit.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+              var baseTag = testDiv.getElementsByTagName( "base" )[ 0 ];
+
+              equal( baseTag.href, expectedBaseSrc, "Generated expected base tag" );
+
+              testDiv.innerHTML = bodyString;
+
+              var popcornScripts = testDiv.getElementsByTagName( "script" )[ 0 ],
+                  butterCleanDiv = testDiv.querySelector( "#test" );
+
+              ok( !butterCleanDiv.hasAttribute( "butter-clean" ), "Removed the butter-clean attribute" );
```
Comment (secretrobotron) [inline test/page/page.unit.js]: I think this stuff is really good. Perhaps you can use querySelectors to run the three `data-butter-*` tests that you run above, right?

Comment (mjschranz) [inline test/page/page.unit.js]: I don't think so. Doesn't querySelector cover specifically css/ids? I'm checking that those attributes were removed.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b2203573a90cecc/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/8dd89d371c6407e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/b2203573a90cecc/output.txt

Total script time: 0.68 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/8dd89d371c6407e/output.txt

Total script time: 0.99 mins

Server is running on http://pr629.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a0804f9a70781de/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a0804f9a70781de/output.txt

Total script time: 0.72 mins

Server is running on http://pr629.butyr.org/templates/test.html

File: test/butter.testrunner.js [hunk 15]
```diff
@@ -1,151 +1,233 @@
...
+              totalTime,
+              ' milliseconds.<br/>',
+              '<span class="passed">',
+              totalPass,
+              '</span> tests of <span class="total">',
+              totalRun,
+              '</span> passed, <span class="failed">',
+              totalFail,
+              '</span> failed.'
+            ].join('');
```
Comment (ScottDowne) [inline test/butter.testrunner.js:149]: The above is all in single quotes, the rest of the file is double quotes.

Change the above to double quotes.

Comment (ScottDowne) [inline test/butter.testrunner.js:149]: Ah, I see why now, we have double quotes in the string. Hm.

**Pull request merged**